app/scripts/modules/core/pipeline/config/stages/webhook/webhookStage.ts

```diff
@@ -56,6 +59,23 @@ export class WebhookStage {
     this.stage.statusUrlResolution = this.viewState.statusUrlResolution;
   }
 
+  public removeCustomHeader(key: string): void {
+    delete this.stage.customHeaders[key];
+  }
+
+  public addCustomHeader(): void {
+    if (!this.stage.customHeaders) {
+      this.stage.customHeaders = {};
+    }
+    const that = this;
+    this.$uibModal.open({
+      templateUrl: require('core/pipeline/config/stages/webhook/modal/addCustomHeader.html'),
+      controller: 'WebhookStageAddCustomHeaderCtrl',
+      controllerAs: 'addCustomHeader',
+    }).result.then(function(customHeader: any) {
+      that.stage.customHeaders[customHeader.key] = customHeader.value;
```

Comment:
I was looking how to deal with that ugly `that`, will do your suggestion! Thanks!